frontend/android/app/src/main/kotlin/app/ehrenamtskarte/MainActivity.kt

```diff
@@ -1,13 +1,32 @@
 package app.ehrenamtskarte
 
 import android.content.pm.PackageManager
+import android.location.LocationManager
 import android.os.Bundle
-import android.preference.PreferenceManager
 import android.util.Log
+import androidx.annotation.NonNull
 import io.flutter.embedding.android.FlutterActivity
+import io.flutter.embedding.engine.FlutterEngine
+import io.flutter.plugin.common.MethodChannel
 import java.io.File
 
-class MainActivity: FlutterActivity() {
+class MainActivity : FlutterActivity() {
+    private val CHANNEL = "app.ehrenamtskarte/location"
+
+    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
+        super.configureFlutterEngine(flutterEngine)
+        MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler {
+                call, result ->
+            if (call.method == "isLocationServiceEnabled") {
+                val locationManager = context.getSystemService(LOCATION_SERVICE) as LocationManager
+                val gpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER)
+                val networkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
```

Comment:
Why do we have to check for network here?

Reply:
This was copy pasted from some stackoverflow. Ill review it again though.

Reply:
This is not  a network check, but checks whether the network provider is enabled: https://developer.android.com/reference/android/location/LocationManager